src/controllers/accomodation.controller.js

```diff
@@ -70,6 +70,55 @@ class Accommodation {
     const query = await accomodationServices.bookAccommodation(req.user.id, req.body.accommodationId, req.body.roomId, req.body.departureDate, req.body.checkoutDate);
     return response.successMessage(res, 'Booking was successfully processed', 201, query);
   }
+
+  /**
+   * This function helps user to rate accommodation which he booked.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async rateAccomodation(req, res) {
+    try {
+      const { rate } = req.body;
+      const userId = req.user.id;
+      const { accomodationId } = req.params;
+
+      const rateData = { rate, userId, accomodationId };
+
+      const rateAccomodation = await accomodationServices.CreateAccomodationRate(rateData);
+      return response.successMessage(res, 'You rated this accomodation successfully', 201, rateAccomodation);
+    } catch (error) {
+      return response.errorMessage(res, error.message, 500);
+    }
+  }
+
+  /**
+   * This function helps user to rate accommodation which he booked.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async GetAverageAccommodationRate(req, res) {
```

Comment:
@mujohn26 the function name suggests something else. It's as it is getting another type of rate. You can name it getAverageRatings, since it will be inside the accommodation, that suggests its average ratings of the accommodation. Kindly follow that pattern elsewhere applicable.